src/repr/src/row.rs

```diff
@@ -1218,10 +923,21 @@ impl RowPacker {
         })
     }
 
-    /// # Safety
+    /// Returns a copy of this `Row`, clearing the data but not the allocation
+    /// in `self`.
     ///
+    /// The intent is that `self`'s allocation can be used to pack additional
+    /// rows, to reduce the amount of interaction with the allocator.
+    pub fn finish_and_reuse(&mut self) -> Row {
```

Comment:
I wonder if this wants a new name, like maybe `clone_and_clear()` or something that makes more sense on a row vs a row builder? Not pressing, just thinking out loud.